walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -76,8 +71,58 @@
         }
       }
     }
+  }
+
+  private Collection<TimeRange> getAvailableTimesWithMaxOptional(Collection<TimeRange> availableTimes, Collection<Event> events, Collection<String> attendees, long duration){
+    if (attendees.isEmpty()){
+      return availableTimes;
+    }
+    // Overview: There's an ArrayList containing HashSets. All TimeRanges start off in the 
+    // HashSet in the first index. For each event/attendee combination, if it doesn't conflict
+    // with a TimeRange, move that TimeRange up an index. If it partially conflicts, split
+    // the TimeRange and move up the non-conflicting TimeRange. 
+    int size = attendees.size()*events.size() + 1;
```

Comment:
I originally had it at just attendees.size()+1, but then I realized that if one attendee is scheduled for multiple events, there would be an indexing error. Ideally, this is something that I'd try to change so that only attendees matter and not attendee/event pairs (because e.g. if one attendee has five events scheduled at the same time and two attendees have an event scheduled together at a different time, the slot with the two attendees would still be chosen over the slot with one attendee and five events, which shouldn't be the case).

Reply:
I see what you mean, thanks for the explanation